### PR TITLE
Refactor functions docs

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

Refactor some of function docs, so they're easier to digest
## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/routes/docs/products/functions/quick-start/+page.markdoc [hunk 1]
```diff
@@ -46,12 +46,26 @@ The easiest way to execute your first function is to use the Appwrite Console.
...
+{% cards_item href="/docs/products/functions/development#response" title="Response" %}
+Learn about function response types, like text, json, html, redirect, and empty responses.
+{% /cards_item %}
+{% cards_item href="/docs/products/functions/deployment" title="Deploy" %}
+Learn to deploy Appwrite Functions from Git, the CLI, or manual uploads.
+{% /cards_item %}
+{% cards_item href="/docs/products/functions/execution" title="Execute" %}
+Appwrite Functions can be executed in a variety of ways, like SDK, domains, schedules, events, and more. Learn about executions.
+{% /cards_item %}
+{% cards_item href="/docs/products/functions/execution#domains" title="Domains" %}
```
Comment (Meldiron) [inline src/routes/docs/products/functions/quick-start/+page.markdoc:62]: CleanShot 2024-02-02 at 13 45 50@2x

I tried and most of the links don't work

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,185 @@
+---
+layout: article
+title: Functions
+description: Learn what an Appwrite Function can do for you and how to create a new Appwrite Function
+---
+
+Each Appwrite Function is a piece of user defined code that can be executed on demand.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:7]: I wouldnt call it user-defined. either deveoper-defined, or defined by you.
User-defined sounds like end user

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+When you create a new Appwrite Function, you select a name, ID, and [runtime language](/docs/products/functions/runtimes).
+
+Each time a function changes, a [deployment](/docs/products/functions/deployments) is created, which is like a version of a function.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:10]:
```suggestion
Each time a function's code changes, a [deployment](/docs/products/functions/deployments) is created, which is like a version of a function.
```

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  tar --exclude code.tar.gz -czf code.tar.gz .
+  ```
+{% /multicode %}
+
+Upload your `.tar.gz` file and specify the entry point of your function, in this case `index.js`.
+Remember to specify the build commands for your function to install dependencies.
+{% /tabsitem %}
+
+{% tabsitem #cli title="CLI" %}
+You can create functions using the CLI, without needing to access the console.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:69]:
```suggestion
You can create functions using the CLI, without needing to access the Console.
```

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+{% /only_dark %}
+{% only_light %}
+![Create project screen](/images/docs/functions/template.png)
+{% /only_light %}
+{% /tabsitem %}
+
+{% tabsitem #manual title="Manual" %}
+You can also create Appwrite Functions manually by uploading your code in a zipped file.
+In the **Create Function** modal, click **create a function manually** at the bottom to switch to manual create wizard.
+You will asked to upload a zip file with your code.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:39]:
```suggestion
You will be asked to upload a zip file with your code.
```

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+First, navigate inside the folder that contains your dependency file. 
+For example, when you list the content of your folder for a Node.js function, it will look like this.
+
+```text
+.
+├── package.json
+└── index.js
+```
+
+Package your code files into the `.tar.gz` format. **Don't include your dependencies folder**, such as **node_modules**.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:50]:
```suggestion
Package your code files into the `.tar.gz` format. **Don't include your dependencies folder**, such as node_modules.
```
Too much boldness, this makes it better

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+You can create Appwrite Functions in three different ways.
+
+{% tabs %}
+{% tabsitem #git title="Git" %}
+It's recommended to create functions that are connected to version control. 
+This lets you track your code using Git, which makes it easy to integrate Appwrite Functions into your existing code base.
+1. In the Appwrite Console's sidebar, click **Functions**.
+2. Click **Create function**.
+3. Connect your project to a Git repository. Your functions will build and deploy when the repository receives a push.
+4. If you already have a repository containing an Appwrite Function, connect to it under **Connect Git repository**.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:25]:
```suggestion
4. If you already have a repository containing an Appwrite Function, select it under **Connect Git repository**. 
```

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+Every deployment of a function uses the same runtime. 
+You can update the runtime of a function by navigating to function > **Settings** > **Runtime**.
+Select a new runtime and click **Update**.
+
+{% info title="Redeployment required" %}
+This change requires your function to be redeployed to take effect.
+{% /info %}
+
+When you update a function's runtime, the code of your function may not be compatible with the new runtime.
+Remember to update your code and redeploy your function.
```
Comment (Meldiron) [inline src/routes/docs/products/functions/functions/+page.markdoc:107]:
```suggestion
When you update a function's runtime, you may need to update code of your function to make it compatible with the new runtime.
```

Above we already have big alert box teiing to redeploy.

File: src/routes/docs/products/functions/develop/+page.markdoc
Comment (PineappleIOnic) [inline src/routes/docs/products/functions/develop/+page.markdoc:1]: We probably want to add something to the `Request` section here noting that `req.body` will contain the object of the event that invoked it, like for `buckets.*.files.*.create` it'll contain the file that triggered that event

File: static/images/docs/platform/execution-details.png
[the remark this answers is not in the document]
Comment (gewenyu99) [reply static/images/docs/platform/execution-details.png:1]: I added it to an issue. It is

File: src/routes/docs/products/functions/execute/+page.markdoc [hunk 9]
```diff
@@ -464,7 +464,7 @@ You'll be able to mock executions by configuring the path, method, headers, and
 {% /only_light %}
 
 
-## Events {% #events %}
+# Events {% #events %}
 
 Changes in Appwrite emit events. You can configure Functions to be executed in response to these events.
```
Comment (PineappleIOnic) [inline src/routes/docs/products/functions/execute/+page.markdoc:469]: Same thing here for the events section, we should explain that the event trigger can be found in `x-appwrite-event` header and the body will contain the data of the resource that triggered it

File: src/routes/docs/products/functions/quick-start/+page.markdoc (hunk 1 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/quick-start/+page.markdoc:62]: Oh yeah I broke all these.

File: src/partials/auth-events.md [hunk 10]
```diff
@@ -1,32 +1,144 @@
...
+---
+* `users.*.verification.*.create`
+* 
+    This event triggers when a verification token for a user is created.
+    Returns [Token Object](/docs/references/cloud/models/token)
+---
+* `users.*.verification.*.update`
+* 
+    This event triggers when a verification token for a user is validated.
+    Returns [Token Object](/docs/references/cloud/models/token)
```
Comment (PineappleIOnic) [inline src/partials/auth-events.md:143]: We need to make sure we have the 1.5 events

File: src/routes/docs/products/functions/develop/+page.markdoc [hunk 11]
```diff
@@ -12,11 +12,14 @@ There is a clear lifecycle for all Appwrite Functions, from beginning to end.
 Here's everything that happens during a function execution.
 
 1. The function is invoked.
-2. Appwrite passes in request information like headers, body or path through the `context.req` object.
-3. The runtime executes the code you defined, you can log through the `context.log()` or `context.error()` methods.
-4. Function terminates when you return results using `return context.res.send()`, `return context.res.json()` or similar.
+1. The active [deployment](/docs/products/functions/deploy)'s executor will handle the request.
```
Comment (PineappleIOnic) [inline src/routes/docs/products/functions/develop/+page.markdoc:15]: These steps are all 1.

File: src/routes/docs/products/functions/domains/+page.markdoc [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+You can use common practices like using paths, query parameters, headers, HTTP methods, formdata, 
+and all the typical HTTP concepts to implement Appwrite Functions.
+
+{% arrow_link href="/docs/products/functions/develop" %}
+Learn about Function development
+{% /arrow_link %}
+
+# Generated domains {% #generated-domains %}
+
+1. In the Appwrite Console's sidebar, click **Functions**.
```
Comment (PineappleIOnic) [inline src/routes/docs/products/functions/domains/+page.markdoc:18]: These steps are all 1.

File: src/routes/docs/products/functions/execute/+page.markdoc [hunk 13]
```diff
@@ -12,34 +12,35 @@ Executions can be invoked through the Appwrite SDK and visiting its REST endpoin
 Here are all the different ways to consume your Appwrite Functions.
 
 # Domains {% #domains %}
-
+You can execute a function through HTTP requests, using a browser or by sending an HTTP request.
 1. In the Appwrite Console's sidebar, click **Functions**.
-1. Under the **Domains** tab, you'll find the generated domain from Appwrite and your custom domains.
+1. Under **Execute access**, set the access to `Any` so that anyone can execute the function. You will use [JWTs](/docs/products/auth/jwt) to authenticate users.
+1. Under the **Domains** tab, you'll find the generated domain from Appwrite and your custom domains. [Learn about adding a custom domain](/docs/products/functions/domains).
```
Comment (PineappleIOnic) [inline src/routes/docs/products/functions/execute/+page.markdoc:18]: These steps are both 1.

File: src/routes/docs/products/functions/templates/+page.markdoc [hunk 14]
```diff
@@ -0,0 +1,128 @@
+---
+layout: article
+title: Templates
+description: Learn about Appwrite Functions' templates that let you jump start function development to extend your Appwrite APIs.
```
Comment (PineappleIOnic) [inline src/routes/docs/products/functions/templates/+page.markdoc:4]: Could the `'` in this description cause problems with the parser? I vaguely remember an issue like this, please check it doesn't result in a server error

**Review by PineappleIOnic — CHANGES_REQUESTED** [review]
Small issues here and these

File: src/routes/docs/products/functions/deployments/+page.markdoc
Comment (loks0n) [inline src/routes/docs/products/functions/deployments/+page.markdoc:1]: Looks very similar content to the `deploy` concept page? Can we just use this for that instead?

File: src/routes/docs/products/functions/develop/+page.markdoc [hunk 15]
```diff
@@ -214,8 +217,8 @@ Future main(final context) async {
   //
   // final client = Client()
   //    .setEndpoint('https://cloud.appwrite.io/v1')
-  //    .setProject(Platform.environment["APPWRITE_FUNCTION_PROJECT_ID"])
-  //    .setKey(Platform.environment["APPWRITE_API_KEY"]);
+  //    .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
```
Comment (loks0n) [inline src/routes/docs/products/functions/develop/+page.markdoc:220]: This should be reverted

File: src/routes/docs/products/functions/develop/+page.markdoc [hunk 16]
```diff
@@ -1088,7 +1091,6 @@ You can access these logs through the following steps.
 
 # Accessing environment variables {% #environment-variables %} 
 If you need to pass constants or secrets to Appwrite Functions, you can use environment variables. 
-Environmental variables can be global, or function-specific.
```
Comment (loks0n) [inline src/routes/docs/products/functions/develop/+page.markdoc:1091]: Why remove this?

File: src/routes/docs/products/functions/domains/+page.markdoc [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+1. Navigate to the Appwrite Console's **Functions** page.
+2. Navigate to the **Domains** tab.
+3. Click on **Create domain**.
+4. Input your domain and click **Next**.
+5. Copy the **CNAME** record and add it to your domain registrar.
+6. Click **Go to console** and wait for verification and certificate generation.
+
+DNS records can take up to 48 hours to propagate.
+
+Appwrite generates TLS certificates to enforce HTTPS on all Appwrite Functions domains, generated or custom.
```
Comment (loks0n) [inline src/routes/docs/products/functions/domains/+page.markdoc:40]: This is the same as L26 - L27 ? Repeated

File: src/routes/docs/products/functions/execute/+page.markdoc [hunk 18]
```diff
@@ -58,7 +59,7 @@ try {
     const execution = await functions.createExecution(
         '[FUNCTION_ID]',
         JSON.stringify({ 'foo': 'bar' }),
-        false,
+        false, // if the function should be executed asynchronously
         '/',
         'GET',
         { 'X-Custom-Header': '123' }
```
Comment (loks0n) [inline src/routes/docs/products/functions/execute/+page.markdoc:65]: Should we add comments to label each parameter?

File: src/routes/docs/products/functions/execute/+page.markdoc [hunk 19]
```diff
@@ -160,15 +161,15 @@ import { Client, Functions } from 'node-appwrite';
...
     .setProject('<PROJECT_ID>')
-    .setKey('[API_KEY]');
+    .setKey('<API_KEY>');
 
 const functions = new Functions(client);
 
 try {
     const execution = await functions.createExecution(
         '[FUNCTION_ID]',
         JSON.stringify({ 'foo': 'bar' }),
```
Comment (loks0n) [inline src/routes/docs/products/functions/execute/+page.markdoc:171]: Should we add more label comments here?

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+
+{% info title="Redeployment required" %}
+This change requires your function to be redeployed to take effect.
+{% /info %}
+
+Under **Git settings** you can configure the Git repository and branch that your function is connected to.
+
+Build commands and entrypoint are executed relative to the configured **Root directory** of your Git respository.
+
+By default, Appwrite will create comments in PRs to your connected branch. You can use **Silent mode** to suppress these comments.
```
Comment (loks0n) [inline src/routes/docs/products/functions/functions/+page.markdoc:132]: Should we explain where/how silent mode is enabled?

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 2 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Addressed

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 3 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: addressed

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 4 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: addressed

File: src/routes/docs/products/functions/execute/+page.markdoc (hunk 9 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/execute/+page.markdoc:470]: Done

File: src/routes/docs/products/functions/develop/+page.markdoc
Comment (gewenyu99) [reply src/routes/docs/products/functions/develop/+page.markdoc:1]: Done

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 8 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Done

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 21]
```diff
@@ -0,0 +1,185 @@
...
+# Create function {% #create-function %}
+You can create Appwrite Functions in three different ways.
+
+{% tabs %}
+{% tabsitem #git title="Git" %}
+It's recommended to create functions that are connected to version control. 
+This lets you track your code using Git, which makes it easy to integrate Appwrite Functions into your existing code base.
+1. In the Appwrite Console's sidebar, click **Functions**.
+2. Click **Create function**.
+3. Connect your project to a Git repository. Your functions will build and deploy when the repository receives a push.
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Done

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 7 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Done

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 6 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Done

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 5 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Done

File: src/routes/docs/products/functions/functions/+page.markdoc [hunk 22]
```diff
@@ -0,0 +1,185 @@
...
+The init command will initialize a folder with a starter function code.
+To deploy the generated code, add dependencies, and deploy the function using the following command:
+
+```bash
+appwrite deploy function
+```
+{% /tabsitem %}
+{% /tabs %}
+
+After creating your function, create a [deployment](/docs/products/functions/deployments) to add code to the function.
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc]: Done

File: src/routes/docs/products/functions/develop/+page.markdoc (hunk 11 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/develop/+page.markdoc:15]: Yes markdown renders them as 1.2.3.4 ;) This in fact works

File: src/partials/auth-events.md (hunk 10 above)
Comment (gewenyu99) [reply src/partials/auth-events.md:143]: Oi, review this:  https://github.com/appwrite/appwrite/pull/7744

File: src/routes/docs/products/functions/domains/+page.markdoc (hunk 12 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/domains/+page.markdoc:18]: Yes markdown renders them as 1.2.3.4 ;) This in fact works

File: src/routes/docs/products/functions/execute/+page.markdoc (hunk 13 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/execute/+page.markdoc:18]: Yes markdown renders them as 1.2.3.4 ;) This in fact works

File: src/routes/docs/products/functions/templates/+page.markdoc (hunk 14 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/templates/+page.markdoc:4]: I'll check :)

File: src/routes/docs/products/functions/deployments/+page.markdoc
Comment (gewenyu99) [reply src/routes/docs/products/functions/deployments/+page.markdoc:1]: Done

File: src/routes/docs/products/functions/develop/+page.markdoc (hunk 16 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/develop/+page.markdoc:1091]: This is not about configuring these variables, just consuming it. No reason to talk about it here, the include steps are the exact same.

File: src/routes/docs/products/functions/execute/+page.markdoc (hunk 19 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/execute/+page.markdoc:171]: Added

File: src/routes/docs/products/functions/functions/+page.markdoc (hunk 20 above)
Comment (gewenyu99) [reply src/routes/docs/products/functions/functions/+page.markdoc:132]: The entire section, when read as a whole, talks about these Git settings together, should be fine.

**Review by loks0n — APPROVED** [review]

**Review by PineappleIOnic — APPROVED** [review]

**Pull request merged**